### PR TITLE
Allow recursive dca picker in the same table

### LINKED ISSUE #883: Seitenstruktur: Interne Weiterleitung zeigt Modal ohne Radio-Buttons
Wenn ich eine interne Weiterleitung anlegen will, fehlen die Radio-Buttons, um eine Seite auszuwählen, siehe Screenshot. 
modal-interne-weiterleitung

### PR DESCRIPTION
**Pull request by aschempp**
fixes #883

Unfortunately, there's no cleaner solution…

Also tested to edit a page within the page picker and pick another page for the redirect 💪

### CODE REVIEW

File: src/Resources/contao/classes/DataContainer.php [hunk 1]
```diff
@@ -874,14 +874,21 @@ protected function generateGlobalButtons()
...
+			return false;
+		}
+
+		list($this->strPickerTable, $this->strPickerField, $this->intPickerId) = explode('.', \Input::get('target'), 3);
+		\Input::setGet('target', null);
+
 		$menuBuilder = \System::getContainer()->get('contao.menu.picker_menu_builder');
 
-		if (!$menuBuilder->supportsTable($this->strTable))
+		if (!$menuBuilder->supportsTable($this->strTable) || empty($this->strPickerTable) || empty($this->strPickerField) || empty($this->intPickerId))
```
Comment (leofeyer) [inline src/Resources/contao/classes/DataContainer.php]: We should run those checks before we initialize the menu builder, shouldn't we?

Comment (aschempp) [inline src/Resources/contao/classes/DataContainer.php]: Agree, changed in 4665cd5cd7c82dc688726cfbcfba35288c29913a

**Comment by leofeyer** [comment]
For some reason, this change breaks the link picker (TinyMCE).

**Comment by leofeyer** [comment]
It is the `empty($this->intPickerId)` part. Why must the picker ID be set?

**Comment by leofeyer** [comment]
Also, the picker is not applied correctly anymore now:



There should be no yellow background in the news archive view.

**Comment by leofeyer** [comment]
Reverted and fixed in 887bac78b8cf12a02864a97bab8a724324356f70 now.

**Pull request merged**